pkg/apis/crd/v1alpha1/types.go

```diff
@@ -709,3 +709,152 @@ type ExternalNodeList struct {
+	// to process the request or gets timeout.
+	BundleFailed BundlePhase = "Failed"
+)
+
+// BundleFailedReason defines the reason why a SupportBundleCollection is marked as Failed.
+type BundleFailedReason string
+
+const (
+	// PartialSuccess means only some required Nodes or ExternalNodes succeed to process
+	// the SupportBundleCollection request before expiration, and others are failed.
+	PartialSuccess BundleFailedReason = "Partial success"
+	// ControllerError means an error occurs on Antrea Controller which breaks processing
+	// the SupportBundleCollection.
+	ControllerError BundleFailedReason = "Controller error"
+	// UnknownError means an unknown error occurs in Antrea when processing the
+	// SupportBundleCollection
+	UnknownError BundleFailedReason = "Unknown"
+)
+
+type SupportBundleStatus struct {
+	Phase BundlePhase `json:"phase"`
+	// The generation observed by Antrea.
+	ObservedGeneration int64 `json:"observedGeneration"`
+	// The time from which the support bundle request is started to be processed by Antrea.
+	StartTime metav1.Time `json:"startTime"`
+	// The number of Nodes and ExternalNodes that have processed the SupportBundleCollection.
+	CurrentNodes int32 `json:"currentNodes"`
+	// The total number of Nodes and ExternalNodes that should process the SupportBundleCollection.
+	DesiredNodes int32 `json:"desiredNodes"`
+	// The reason why the SupportBundleCollection phase is marked as Failed.
+	FailedReason BundleFailedReason `json:"failedReason,omitempty"`
+}
+
+type BundleNodes struct {
+	// List the names of certain Nodes which are expected to collect and upload
+	// bundle files.
+	// +optional
+	NodeNames []string `json:"nodeNames,omitempty"`
+	// Select certain Nodes which match the label selector.
+	// +optional
+	NodeSelector *metav1.LabelSelector `json:"nodeSelector,omitempty"`
+}
+
+type BundleExternalNodes struct {
+	Namespace string
+	// List the names of certain ExternalNodes which are expected to collect and upload
+	// bundle files.
+	// +optional
+	NodeNames []string `json:"nodeNames,omitempty"`
+	// Select certain ExternalNodes which match the label selector.
+	// +optional
+	NodeSelector *metav1.LabelSelector `json:"nodeSelector,omitempty"`
+}
+
+// BundleUploadScheme defines the scheme that the BundleFileServer uses in the url to
+// upload bundle files.
+type BundleUploadScheme string
+
+const (
+	HTTPScheme  BundleUploadScheme = "http"
+	HTTPSScheme BundleUploadScheme = "https"
+)
+
+// BundleUploadVerb defines the verb to upload bundle files to the URL provided by
+// the BundleFileServer.
+type BundleUploadVerb string
+
+const (
+	PostBundle BundleUploadVerb = "post"
+	PutBundle  BundleUploadVerb = "put"
+)
+
+// BundleFileServer specifies the bundle file server information.
+type BundleFileServer struct {
+	Host   string             `json:"host"`
+	Path   string             `json:"path"`
+	Scheme BundleUploadScheme `json:"scheme"`
+	Verb   BundleUploadVerb   `json:"verb"`
```

Comment:
This is only for HTTP right. Do we really need to support both PUT and POST? Could not we just choose one?